src/corelibs/U2Gui/src/util/U2FileDialog.cpp

```diff
@@ -94,11 +108,68 @@ QString U2FileDialog::getSaveFileName(QWidget* parent,
                                       const QString& caption,
                                       const QString& dir,
                                       const QString& filter,
-                                      QString* selectedFilter,
+                                      const QString& selectedFilter,
                                       const QFileDialog::Options& options) {
     activateAppWindow();
     QFileDialog::Options effectiveOptions = getEffectiveOptions(options);
-    return QFileDialog::getSaveFileName(parent, caption, dir, filter, selectedFilter, effectiveOptions);
+    return U2FileDialog::getFileName(parent,
+                                     caption,
+                                     dir,
+                                     filter,
+                                     selectedFilter,
+                                     effectiveOptions,
+                                     QFileDialog::AcceptSave,
+                                     QFileDialog::ExistingFile);
 }
 
+QStringList U2FileDialog::getFileNames(QWidget* parent,
+                                       const QString& caption,
+                                       const QString& dir,
+                                       const QString& filter,
+                                       const QString& selectedFilter,
+                                       QFileDialog::Options options,
```

Comment:
Passing primitive types (`enum` is, basically, just an usual `int`) by the const reference doesn't make any sense and, even, incorrect, because it causes some complications (this was asked and answered numerous amount of times, like [here](https://stackoverflow.com/a/10770431/7273346), [here](https://stackoverflow.com/a/4112947/7273346), [here](https://stackoverflow.com/a/14013189/7273346) and [here](https://stackoverflow.com/a/6834608/7273346)). I have never done this and see no point to begin.

Reply:
> these are 3rd party type and a reader is not aware if they are classes or enums

Actually, this is the reason why it should be without `const &` - to aware reader, that this is not a class

Reply:
We need a common style for the whole application: please update to `const T&`

Reply:
OK